soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -12620,12 +12627,14 @@ s32 func_8084DFF4(GlobalContext* globalCtx, Player* this) {
         }
 
         Message_StartTextbox(globalCtx, giEntry.textId, &this->actor);
-        if (giEntry.modIndex == MOD_NONE) {
-            Item_Give(globalCtx, giEntry.itemId);
-        } else {
-            Randomizer_Item_Give(globalCtx, giEntry);
+        if (giEntry.modIndex != MOD_RANDOMIZER && giEntry.itemId != RG_ICE_TRAP) {
```

Comment:
seems this exact check exists in a couple places, might be nice to macro it (another future code cleanup thing, doesn't need to be part of this PR)